File: DTMS365-AppSource/DynamicsTMS365/Models/tblStaticRouteCarr.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using DTO = Ngl.FreightMaster.Data.DataTransferObjects;
using LTS = Ngl.FreightMaster.Data.LTS;
using DTran = Ngl.Core.Utility.DataTransformation;


namespace DynamicsTMS365.Models
{
                 
    public class tblStaticRouteCarr
    {


        //public tblRouteTypeRefLane tblRouteTypeRefLane { get; set; }
       // public tblStaticRoute tblStaticRoute { get; set; }
       // public CarrierRefLane CarrierRefLane { get; set; }
        //public EntitySet<tblStaticRouteEquip> tblStaticRouteEquips { get; set; }
        public string StaticRouteCarrURI { get; set; }
        public string StaticRouteStateFilter { get; set; }
        //public Binary StaticRouteCarrUpdated { get; set; }
        public string StaticRouteCarrModUser { get; set; }
        public DateTime? StaticRouteCarrModDate { get; set; }
        public bool StaticRouteCarrAutoAcceptLoads { get; set; }
        public bool StaticRouteCarrRequireAutoTenderApproval { get; set; }
        public int StaticRouteCarrRouteSequence { get; set; }
        public int StaticRouteCarrTransType { get; set; }
        public bool StaticRouteCarrHazmatFlag { get; set; }
        public int StaticRouteCarrMaxStops { get; set; }
        public int StaticRouteCarrTendLeadTime { get; set; }
        public bool StaticRouteCarrAutoTenderFlag { get; set; }
        public int StaticRouteCarrRouteTypeCode { get; set; }
        public string StaticRouteCarrDescription { get; set; }
        public string StaticRouteCarrName { get; set; }
        public string StaticRouteCarrCarrierName { get; set; }
        public int StaticRouteCarrCarrierNumber { get; set; }
        public int StaticRouteCarrCarrierControl { get; set; }
        public int StaticRouteCarrStaticRouteControl { get; set; }
        public int StaticRouteCarrControl { get; set; }


        //public Binary StaticRouteCarrUpdated { get; set; }


        private byte[] _StaticRouteCarrUpdated;
        /// <summary>
        /// StaticRouteCarrUpdated should be bound to UI, _StaticRouteCarrUpdated is only bound on the controller
        /// </summary>
        public string StaticRouteCarrUpdated
        {
            get
            {
                if (this._StaticRouteCarrUpdated != null) { return Convert.ToBase64String(this._StaticRouteCarrUpdated); }
                return string.Empty;
            }
            set
            {
                if (string.IsNullOrEmpty(value)) { this._StaticRouteCarrUpdated = null; } else { this._StaticRouteCarrUpdated = Convert.FromBase64String(value); }
            }
        }

        public void setUpdated(byte[] val) { _StaticRouteCarrUpdated = val; }
        public byte[] getUpdated() { return _StaticRouteCarrUpdated; }


        public static Models.tblStaticRouteCarr selectModelData(LTS.tblStaticRouteCarr d)
        {
            Models.tblStaticRouteCarr modelRecord = new Models.tblStaticRouteCarr();
            if (d != null)
            {
                List<string> skipObjs = new List<string> { "tblRouteTypeRefLane","tblStaticRoute","CarrierRefLane", "StaticRouteCarrUpdated", "tblStaticRouteEquips" };
                string sMsg = "";
                modelRecord = (Models.tblStaticRouteCarr)DTran.CopyMatchingFields(modelRecord, d, skipObjs, ref sMsg);
                if (modelRecord != null) { modelRecord.setUpdated(d.StaticRouteCarrUpdated.ToArray()); }
            }
            return modelRecord;
        }

        public static LTS.tblStaticRouteCarr selectLTSData(Models.tblStaticRouteCarr d)
        {
            LTS.tblStaticRouteCarr LTSRecord = new LTS.tblStaticRouteCarr();
            if (d != null)
            {
                List<string> skipObjs = new List<string> { "tblRouteTypeRefLane", "tblStaticRoute", "CarrierRefLane", "StaticRouteCarrUpdated", "tblStaticRouteEquips" };
                string sMsg = "";
                LTSRecord = (LTS.tblStaticRouteCarr)DTran.CopyMatchingFields(LTSRecord, d, skipObjs, ref sMsg);
                if (LTSRecord != null)
                {
                    byte[] bupdated = d.getUpdated();
                    LTSRecord.StaticRouteCarrUpdated = bupdated == null ? new byte[0] : bupdated;
                }
            }
            return LTSRecord;
        }
    }
}

File: NGL.DataAssemblies/Ngl.FreightMaster.Data/DTO/LaneData/tblStaticRouteCarr.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Runtime.Serialization
Imports Ngl.Core.ChangeTracker

Namespace DataTransferObjects
    <DataContract()> _
    Public Class tblStaticRouteCarr
            Inherits DTOBaseClass


#Region " Data Members"
        Private _StaticRouteCarrControl As Integer = 0
        <DataMember()> _
        Public Property StaticRouteCarrControl() As Integer
            Get
                Return _StaticRouteCarrControl
            End Get
            Set(ByVal value As Integer)
                _StaticRouteCarrControl = value
            End Set
        End Property

        Private _StaticRouteCarrStaticRouteControl As Integer = 0
        <DataMember()> _
        Public Property StaticRouteCarrStaticRouteControl() As Integer
            Get
                Return _StaticRouteCarrStaticRouteControl
            End Get
            Set(ByVal value As Integer)
                _StaticRouteCarrStaticRouteControl = value
            End Set
        End Property

        Private _StaticRouteCarrCarrierControl As Integer = 0
        <DataMember()> _
        Public Property StaticRouteCarrCarrierControl() As Integer
            Get
                Return _StaticRouteCarrCarrierControl
            End Get
            Set(ByVal value As Integer)
                _StaticRouteCarrCarrierControl = value
            End Set
        End Property

        Private _StaticRouteCarrCarrierNumber As Integer = 0
        <DataMember()> _
        Public Property StaticRouteCarrCarrierNumber() As Integer
            Get
                Return _StaticRouteCarrCarrierNumber
            End Get
            Set(ByVal value As Integer)
                _StaticRouteCarrCarrierNumber = value
            End Set
        End Property

        Private _StaticRouteCarrCarrierName As String = ""
        <DataMember()> _
        Public Property StaticRouteCarrCarrierName() As String
            Get
                Return Left(_StaticRouteCarrCarrierName, 50)
            End Get
            Set(ByVal value As String)
                _StaticRouteCarrCarrierName = Left(value, 50)
            End Set
        End Property

        Private _StaticRouteCarrName As String = ""
        <DataMember()> _
        Public Property StaticRouteCarrName() As String
            Get
                Return Left(_StaticRouteCarrName, 50)
            End Get
            Set(ByVal value As String)
                _StaticRouteCarrName = Left(value, 50)
            End Set
        End Property

        Private _StaticRouteCarrDescription As String = ""
        <DataMember()> _
        Public Property StaticRouteCarrDescription() As String
            Get
                Return Left(_StaticRouteCarrDescription, 255)
            End Get
            Set(ByVal value As String)
                _StaticRouteCarrDescription = Left(value, 255)
            End Set
        End Property

        Private _StaticRouteCarrRouteTypeCode As Integer = 0
        <DataMember()> _
        Public Property StaticRouteCarrRouteTypeCode() As Integer
            Get
                Return _StaticRouteCarrRouteTypeCode
            End Get
            Set(ByVal value As Integer)
                _StaticRouteCarrRouteTypeCode = value
            End Set
        End Property

        Private _StaticRouteCarrAutoTenderFlag As Boolean = False
        <DataMember()> _
        Public Property StaticRouteCarrAutoTenderFlag() As Boolean
            Get
                Return _StaticRouteCarrAutoTenderFlag
            End Get
            Set(ByVal value As Boolean)
                _StaticRouteCarrAutoTenderFlag = value
            End Set
        End Property

        Private _StaticRouteCarrTendLeadTime As Integer = 0
        <DataMember()> _
        Public Property StaticRouteCarrTendLeadTime() As Integer
            Get
                Return _StaticRouteCarrTendLeadTime
            End Get
            Set(ByVal value As Integer)
                _StaticRouteCarrTendLeadTime = value
            End Set
        End Property

        Private _StaticRouteCarrMaxStops As Integer = 0
        <DataMember()> _
        Public Property StaticRouteCarrMaxStops() As Integer
            Get
                Return _StaticRouteCarrMaxStops
            End Get
            Set(ByVal value As Integer)
                _StaticRouteCarrMaxStops = value
            End Set
        End Property

        Private _StaticRouteCarrHazmatFlag As Boolean = False
        <DataMember()> _
        Public Property StaticRouteCarrHazmatFlag() As Boolean
            Get
                Return _StaticRouteCarrHazmatFlag
            End Get
            Set(ByVal value As Boolean)
                _StaticRouteCarrHazmatFlag = value
            End Set
        End Property

        Private _StaticRouteCarrTransType As Integer = 0
        <DataMember()> _
        Public Property StaticRouteCarrTransType() As Integer
            Get
                Return _StaticRouteCarrTransType
            End Get
            Set(ByVal value As Integer)
                _StaticRouteCarrTransType = value
            End Set
        End Property

        Private _StaticRouteCarrRouteSequence As Integer = 0
        <DataMember()> _
        Public Property StaticRouteCarrRouteSequence() As Integer
            Get
                Return _StaticRouteCarrRouteSequence
            End Get
            Set(ByVal value As Integer)
                _StaticRouteCarrRouteSequence = value
            End Set
        End Property

        Private _StaticRouteCarrRequireAutoTenderApproval As Boolean = False
        <DataMember()> _
        Public Property StaticRouteCarrRequireAutoTenderApproval() As Boolean
            Get
                Return _StaticRouteCarrRequireAutoTenderApproval
            End Get
            Set(ByVal value As Boolean)
                _StaticRouteCarrRequireAutoTenderApproval = value
            End Set
        End Property

        Private _StaticRouteCarrAutoAcceptLoads As Boolean = False
        <DataMember()> _
        Public Property StaticRouteCarrAutoAcceptLoads() As Boolean
            Get
                Return _StaticRouteCarrAutoAcceptLoads
            End Get
            Set(ByVal value As Boolean)
                _StaticRouteCarrAutoAcceptLoads = value
            End Set
        End Property

        Private _StaticRouteStateFilter As String
        <DataMember()> _
         Public Property StaticRouteStateFilter As String
            Get
                Return Left(_StaticRouteStateFilter, 255)
            End Get
            Set(value As String)
                _StaticRouteStateFilter = Left(value, 255)
            End Set
        End Property


        Private _StaticRouteCarrModDate As System.Nullable(Of Date)
        <DataMember()> _
        Public Property StaticRouteCarrModDate() As System.Nullable(Of Date)
            Get
                Return _StaticRouteCarrModDate
            End Get
            Set(ByVal value As System.Nullable(Of Date))
                _StaticRouteCarrModDate = value
            End Set
        End Property

        Private _StaticRouteCarrModUser As String = ""
        <DataMember()> _
        Public Property StaticRouteCarrModUser() As String
            Get
                Return Left(_StaticRouteCarrModUser, 100)
            End Get
            Set(ByVal value As String)
                _StaticRouteCarrModUser = Left(value, 100)
            End Set
        End Property

        Private _StaticRouteCarrURI As String = ""
        <DataMember()> _
        Public Property StaticRouteCarrURI() As String
            Get
                Return Left(_StaticRouteCarrURI, 500)
            End Get
            Set(ByVal value As String)
                _StaticRouteCarrURI = Left(value, 500)
            End Set
        End Property

        Private _StaticRouteCarrUpdated As Byte()
        <DataMember()> _
        Public Property StaticRouteCarrUpdated() As Byte()
            Get
                Return _StaticRouteCarrUpdated
            End Get
            Set(ByVal value As Byte())
                _StaticRouteCarrUpdated = value
            End Set
        End Property

#End Region

#Region " Public Methods"
        Public Overrides Function Clone() As DTOBaseClass
            Dim instance As New tblStaticRouteCarr
            instance = DirectCast(MemberwiseClone(), tblStaticRouteCarr)
            Return instance
        End Function

#End Region

    End Class
End Namespace